src/System.Private.ServiceModel/tools/test/Bridge/Bridge/IdleTimeoutManager.cs

```diff
@@ -55,6 +58,26 @@ public IdleTimeoutManager(TimeSpan idleTimeout)
                 _waitTimeout = idleTimeout;
                 _timer = new Timer(this.TimeOutCallback);
                 this.Restart();
+
+                // Register to be notified when the Bridge configuration changes
+                ConfigController.BridgeConfigurationChanged += (object s, Tuple<BridgeConfiguration, BridgeConfiguration> tuple) =>
+                {
+                    BridgeConfiguration newConfig = tuple.Item2;
+
+                    int minutes = newConfig.BridgeIdleTimeoutMinutes;
+                    TimeSpan newTimeout = (minutes <= 0)
```

Comment:
You were looking for some mechanism to use timeout for immediate shutdown. Normally a value of -1 means infinite. If you specify 0 minutes idle, then that would seem to me to mean shutdown immediately and not infinite.